### PR TITLE
Create service agent on post

### PR DESCRIPTION
**Pull request by kilbergr**
## Description

This PR implements the ServiceAgent post endpoint. Creates a `PublicCreateServiceAgentHandler` that creates a service agent on a shipment. Required fields are shipment id, role, and point of contact.
This also includes the creation of definition for Role in the public API.

## Reviewer Notes

I'm fetching the TSP user so I can see whether they own the shipment trying to create the Service Agent. Do you think this auth check is worth the weight of the second fetch? 

## Setup

Use one of the shipment ids from the following to create the service agent on Swagger: http://tsplocal:3000/api/v1/docs#/service_agents/createServiceAgent
```
./bin/generate-test-data -scenario=7
```
You have to be signed in to tsplocal to use.

## Code Review Verification Steps

* [ ] End to end tests pass (`make e2e_test`).
* [x] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [x] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [x] Request review from a member of a different team.
* [x] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/159751437) for this change

### CODE REVIEW

File: pkg/handlers/service_agents_public.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	}
+	return serviceAgentPayload
+}
+
+// PublicCreateServiceAgentHandler ... creates a new service agent on a shipment via POST /shipment/{shipmentId}/serviceAgent
+type PublicCreateServiceAgentHandler HandlerContext
+
+// Handle ... creates a new ServiceAgent from a request payload - checks that currently logged in user is authorized to act for the TSP assigned the shipment
+func (h PublicCreateServiceAgentHandler) Handle(params publicserviceagentop.CreateServiceAgentParams) middleware.Responder {
+	session := auth.SessionFromRequestContext(params.HTTPRequest)
```
Comment (chrisgilmerproj) [inline pkg/handlers/service_agents_public.go]: We've got to check here if its a TSP coming through the right endpoint and that they are logged before we let them create this record on the shipment.  Here's an example I've used for this: https://github.com/transcom/mymove/blob/master/pkg/handlers/shipments_public.go#L71-L91

File: pkg/handlers/service_agents_public.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	payload := params.ServiceAgent
+
+	serviceAgentRole := models.Role(payload.Role)
+	newServiceAgent, verrs, err := models.CreateServiceAgent(
+		h.db,
+		shipment.ID,
+		serviceAgentRole,
+		*payload.PointOfContact,
```
Comment (stangah) [inline pkg/handlers/service_agents_public.go]: How do you know this is not `nil`?

File: pkg/models/service_agent.go [hunk 3]
```diff
@@ -72,3 +75,38 @@ func (s *ServiceAgent) ValidateCreate(tx *pop.Connection) (*validate.Errors, err
...
+		PhoneNumber:      phoneNumber,
+		EmailIsPreferred: emailIsPreferred,
+		PhoneIsPreferred: phoneIsPreferred,
+		Notes:            notes,
+	}
+	verrs, err := tx.ValidateAndCreate(&newServiceAgent)
+	if err != nil || verrs.HasAny() {
+		newServiceAgent = ServiceAgent{}
+	}
+	verrs, err = tx.ValidateAndSave(&newServiceAgent)
```
Comment (stangah) [inline pkg/models/service_agent.go]: Why are you saving a blank ServiceAgent if the initial creation fails?

File: pkg/handlers/service_agents_public.go (hunk 2 above)
Comment (stangah) [reply pkg/handlers/service_agents_public.go]: Hm, I guess it's fine then :)

**Pull request merged**